src/main/java/com/github/bordertech/config/DefaultConfiguration.java

```diff
@@ -870,142 +817,42 @@ private void clearMessages() {
 		messages.setLength(0);
 	}
 
-	/**
-	 * Iterates through the values, looking for values containing ${...} strings. For those that do, we substitute if
-	 * the stuff in the {...} is a defined key.
-	 *
-	 * @return true if any substitutions were made, false otherwise.
-	 */
-	private boolean substitute() {
-		boolean madeChange = false;
-
-		for (String key : backing.keySet()) {
-			madeChange = madeChange || substitute(key);
-		}
-
-		return madeChange;
-	}
-
 	/**
 	 * Performs value substitution for the given key. For values containing ${...} strings, we substitute if the stuff
 	 * in the {...} is a defined key.
 	 *
 	 * @param aKey the key to run the substitution for.
-	 * @return true if a substitutions was made, false otherwise.
 	 */
-	private boolean substitute(final String aKey) {
```

Comment:
It is great to see a lot of the code removed.

Reply:
Yes, I am a big fan of using someone else's code.